cmake/dts.cmake

```diff
@@ -13,10 +13,48 @@ set(GENERATED_DTS_BOARD_CONF ${PROJECT_BINARY_DIR}/include/generated/generated_d
 set_ifndef(DTS_SOURCE ${BOARD_ROOT}/boards/${ARCH}/${BOARD_FAMILY}/${BOARD}.dts)
 set_ifndef(DTS_COMMON_OVERLAYS ${PROJECT_SOURCE_DIR}/dts/common/common.dts)
 
+
+#Parse boards/shields to generate the shield list
+
+set(shield_dir $ENV{ZEPHYR_BASE}/boards/shields)
+
+# Match the .overlay files in the shield directories to make sure we are
+# finding shields, e.g. x_nucleo_iks01a1/x_nucleo_iks01a1.overlay
+file(GLOB_RECURSE shields_refs_list
+     RELATIVE ${shield_dir}
+     ${shield_dir}/*/*.overlay
+     )
+
+# The above gives a list like
+# x_nucleo_iks01a1/x_nucleo_iks01a1.overlay;x_nucleo_iks01a2/x_nucleo_iks01a2.overlay
+# we construct a list of shield names by extracting file name and
+# removing the extension.
+foreach(shield_path ${shields_refs_list})
+  get_filename_component(shield ${shield_path} NAME_WE)
+
+  # Generate CONFIG flags matching each shield
+  string(TOUPPER config_shield_${shield} UPPER_CASE_CONFIG_SHIELD)
+  set(shield_config ${UPPER_CASE_CONFIG_SHIELD})
```

Comment:
It is is cleaner to do this oneliner instead of those two statements:

string(TOUPPER "CONFIG_SHIELD_${shield}" shield_config)

Reply:
Done, thanks